docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,164 @@
+
+### 2.4 Settling for hybrid
+As RAG systems naturally evolve with longer documents, enriched with additional context and more and more uninitiated users, simply transposing their usual search habits, the reality sets in: embedding models are barely competitive to classic information retrieval approaches like BM25 and have to be supplemented by it.
+
+What this means in practice: 
+* Hybrid indexation is emerging as the new standard: on my latest internal benchmarks, combining a keyword-based search with embedding similarity hit the highest retrieval rate (nearly 90% of resources found among the top ten results). The easiest implementation I’ve found so far is lancedb’s but there is right now a lot of activity in this space.
+* Keyword-based search is still a strong alternative. The infrastructure is already there and at a very large scale, embeddification can quickly become prohibitive.
+* Evaluation has to be built from the ground up, based on the constraints of your existing infrastructure (especially in regards to data quality/indexation) and the ways users interact with it.
+
+Overall, there is a lot of alpha left in recovering information retrieval methods and indicators in an LLM context, especially as many key features can be now accurately generated/extracted at scale. Jo Kristian Bergum from Vespa has just introduced a very convincing method of data expansion, by grounding an LLM-as-a-Judge on a small relevant dataset, that can be transformed into a large one. Basically, generative AI does not necessarily replace the classic approaches of retrieval evaluation, but completely reshape their logistics. Intensive data work that would have been only available to large scale organizations is suddenly scalable with much less resources.
+
+## 3. Keep the data alive
+The evolution of RAG toward a communication system means that the data is no longer just a passive reference stock. It has to be kept alive in two ways: continuously transformed and reshaped to better fit the retrieval objective, constantly circulating across different flows.
+
+### 3.1 You need bad data
+The reality of production RAG is that once we get slightly beyond the proof-of-concept phase, the system has to work with whatever data is available. To get to the point, you have to become an expert on bad data.
+
+Scaling is not only a bitter lesson for LLMs: I worked for years in cultural heritage and a library with millions of bad OCR documents is always considerably more useful than a well curated and edited selection of a thousand canon texts. Large organizations are no different. RAG has to work with legacy texts, which means PDFs. And, to stress it again, LLMs are hardly trained on PDFs: common crawl is web/html texts, without artifact and advanced document structures. Bad data is also to be expected on the input side. Users want the best answers with the minimum efforts, and the customer is always right. 
+
```

Comment:
you mean a **non-OCRed selection** of 1000 canonical texts?

Reply:
1. yes. this actually happened a lot in humanities: people would invest lots of money for very clean digital editions of major texts, only for actual users using way more the big collections of millions of uncleaned texts.
2. honestly we don't know. Maybe a better phrasing: "For all that we know, LLM are mostly trained on web/html archives and will have less familiarity with digitization artifacts and advanced document structures met in PDF documents."